integration-tests/src/test/java/org/apache/fineract/integrationtests/AdvancedPaymentAllocationLoanRepaymentScheduleTest.java

```diff
@@ -4113,6 +4113,45 @@ public void uc137() {
         });
     }
 
+    // UC138: Advanced payment allocation, with DownPayment enabled in Loan Product but disabled in Loan application
```

Comment:
Please add a test case for update loan application!
Please add negative test cases to check enabling down payment is not allowed if it was not allowed on the loan product!

Reply:
Done! ITC added